### ISSUE TITLE
New quest: leaf_type for landuse:forest

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#366]
The [`leaf_type`](https://wiki.openstreetmap.org/wiki/Key:leaf_type) quest for forests should be easy to answer and always stays the same… And one quest can cover a big area… :smiley:

**Comment by matkoniecz** [comment]
I intend to make this as the next step after #338

At this moment my main block is a lack of three photos that are

- on suitable license (PD, CC0, GPLv3, CC-BY-SA 4.0 or other GPLv3 compatible)
- working in small resolutions

And three images are required

- one shows needles (and do not suggest other attributes like "young leaves" or otherwise imply that difference is not based solely on leaf type - for example https://github.com/westnordost/StreetComplete/pull/338#issuecomment-309786762 would imply that this quest is about tree shape)
- one shows broad leaves (with the same limitations)
- one shows both (maybe composite of both images above?)

**Comment by rugk** [comment]
As for a "usual leaves" you could also use:
* https://openclipart.org/detail/70483/maple-leaves
* https://openclipart.org/detail/169944/leaves
* https://openclipart.org/detail/170143/eco-green-leaves-icon
* https://openclipart.org/detail/173827/leaves-and-branches-2

As for needles:
* https://commons.wikimedia.org/wiki/File:Juniper_needles.jpg
* https://commons.wikimedia.org/wiki/File:Branched_juniper_twig_with_shots.jpg
* https://commons.wikimedia.org/wiki/File:Pinus_ponderosa_var._scopulorum_(7458435230).jpg
* https://commons.wikimedia.org/wiki/File:Jun_chin_close.jpg
* https://commons.wikimedia.org/wiki/File:Abies_pinsapo_leaves.jpg
* https://commons.wikimedia.org/wiki/File:Taxus-needles.jpg

In real pictures:
* https://commons.wikimedia.org/wiki/File:Tabak_Burley-Jupiter.JPG
* https://commons.wikimedia.org/wiki/File:Playing_with_water,_light_and_leaves_(Fagus_sylvatica).JPG

Or just make it yourself. I could clearly also take some pictures if needed.

**Comment by matkoniecz** [comment]
I really like cliparts idea, that is a new idea. Some nice broadleaved, just https://openclipart.org/search/?query=pine find some needles.

https://commons.wikimedia.org/wiki/File:Branched_juniper_twig_with_shots.jpg would be even better - unfortunately it requires matching broadleaved on black baground

https://commons.wikimedia.org/wiki/File:Tabak_Burley-Jupiter.JPG - I found no evidence that CC-BY-SA 3 is compatible with GPL v3 and strong indication that it is not.

https://commons.wikimedia.org/wiki/File:Playing_with_water,_light_and_leaves_(Fagus_sylvatica).JPG - I considered this one but once in small size it is not very readable. Also, I wanted to keep broad/needle without any spurious differences like colour (red/green)/state (dying/on tree).

**Comment by matkoniecz** [comment]
Thanks for clipart icons idea, this is really promising - especially with readability better than photo at low resolution. The last thing is to check whatever openclipart.org is trustworthy on topic of licenses.

**Comment by westnordost** [comment]
Did you see the trees/leaves in twemoji and emojione icon sets?

**Comment by rugk** [comment]
> unfortunately it requires matching broadleaved on black baground

It's easy to edit it, but could still leave some traces of a black background.

> I found no evidence that CC-BY-SA 3 is usable in GPL v3 project and strong indication that it is not.

As it is media in contrast to code, I think it works. Actually CC-BY-SA is quite the "synonym" to GPLv3. 

> The last thing is to check whatever openclipart.org is trustworthy on topic of licenses.

Public domain/CC0. Look here: https://openclipart.org/share
You won't find any lower criteria. (And I would not suggest images with bad licenses.)

> Did you see the trees/leaves in twemoji and emojione 2 icon sets?

Why not have a general look: http://emojipedia.org/fallen-leaf/
(I also like the EmojiOne 3 version. When not using SVG they are under a free license)

> And now I found also some promising reserve photos at https://commons.wikimedia.org/wiki/Tree?uselang=pl#Foliage_.2F_Feuillage

:+1:

**Comment by westnordost** [comment]
License of EmojiOne 3 is not compatible. Only twemoji (Twitter) and EmojiOne 2 are.

**Comment by rugk** [comment]
Ah, right, because of -NC… Okay, but v2 is not bad either…

**Comment by rugk** [comment]
Any news? Is it really so hard to decide on two pictures?

**Comment by matkoniecz** [comment]
@rugk No, it is because it is a hobby project and as far as StreetComplete goes I was busy with making my own map style for use in StreetComplete what turned out to require for example https://github.com/tilezen/vector-datasource/pulls?q=is%3Apr+author%3Amatkoniecz+is%3Aclosed

**Comment by westnordost** [comment]
Cool

**Comment by rugk** [comment]
Oh, cool. :+1:  So you are working on https://github.com/westnordost/StreetComplete/issues/183. (Always a good idea to leave a note in the issue thee, so no one else starts working in the meantime…)

**Comment by PanderMusubi** [comment]
Good collection of SVG icons is https://github.com/Xaviju/inkscape-open-symbols

Values to offer in case of `natural=tree` are:
* `leaf_type=broadleaved` for broad leaved trees. This includes palm trees which might need to be emphasized or even offered as an extra option, mapping to the same value, to prevent confusion and people tagging palm trees as needle leaved.
* `leaf_type=needleleaved` for needle leaved trees, This includes conifers which might need to be emphasized or even offered as an extra option, mapping to the same value, to prevent confusion and people tagging conifers as broad leaved.

Improved icon and picture suggestions:
* icon broad leaved (only the right one) https://openclipart.org/detail/70483/maple-leaves or https://openclipart.org/detail/169944/leaves
* icon broad leaved palm leave https://openclipart.org/detail/233707/sabal-leaf
* icon needle leaved https://openclipart.org/detail/266238/pine-branch-2
* picture broad leaved https://en.wikipedia.org/wiki/File:Lisc_lipy.jpg or https://commons.wikimedia.org/wiki/File:Alnus_incana_rugosa_leaves.jpg or https://commons.wikimedia.org/wiki/File:Quercus_robur.jpg
* picture palm leaf https://upload.wikimedia.org/wikipedia/commons/7/7e/2007-02-28_(57)_Palm_leaf,_Palmblatt.JPG
* picture needle leaved https://commons.wikimedia.org/wiki/File:Jun_chin_close.jpg (this one has also conifer example) or https://commons.wikimedia.org/wiki/File:Branched_juniper_twig_with_shots.jpg or https://commons.wikimedia.org/wiki/File:Fichtennadel.jpg or https://en.wikipedia.org/wiki/File:Abies_grandis_5359.JPG

Note that `leaf_type=mixed` does not apply to trees and is only used in combination with `landuse=*`. Hence this should not be offered for `natural=tree`. Also `leaf_type=leafless` for cacti is not used on `natural=tree`, as these plants are tree-like, but not trees and only used in `landuse=*`.

Do not process `natural=tree_row` as the individual trees in the row can differ. Process **only** the nodes of `natural=tree_row`. Only when all nodes are tagged for `leaf_cycle`, `natural=tree_row` can automatically get a value which may include `leaf_type=mixed`. Per haps this is something for a bot?

Sometimes, trees are dead and have no leaves for the user to identify them. Any suggestions what to do with that? A tree can have an end date, but the moment of survey is not the date to use there.

**Comment by PanderMusubi** [comment]
(done updating the post above)

**Comment by PanderMusubi** [comment]
Perhaps note the user that a single pine tree in a huge forest doesn't make the forest mixed.

**Comment by westnordost** [comment]
The general problem I see with tagging forests via survey is that it is usually a large area.
1. The user cannot be expected to check out every corner of this forest if it is i.e. broadleaved everywhere. Thus, the quality of the data entered by the user is quite low.
2. It is likely that the forest polygon needs to be split up to accurately map the situation. But this app can't do that.
3. This is information that can be more accurately mapped looking at high-quality satellite images

**Comment by PanderMusubi** [comment]
Okay, agree to omit `natural=wood` and `landuse=forest` from this query.

**Comment by rugk** [comment]
So only https://github.com/westnordost/StreetComplete/issues/83? I would be fine with that.

**Comment by PanderMusubi** [comment]
no, forget that one for now. the table in this post is what is relevant, minus the two omissions

**Comment by PanderMusubi** [comment]
so only

| `leaf_type=` | `broadleaved` | `needleleaved` | `mixed` | `leafless` |
|--:|---|---|---|---|
| `natural=tree`         | + | + | - | - |
| `natural=scrub`      | + | + | + | + |
| `barrier=hedge`     | + | + | + | + |

**Comment by PanderMusubi** [comment]
I removed also tree_row, as that can be calculated from the trees making the row. Some tree rows are very very long, so not practical to do in a survey with StreetComplete

**Comment by matkoniecz** [comment]
> I removed also tree_row, as that can be calculated from the trees making the row. 

IIRC you map either individual trees or natural=tree_row line.

**Comment by matkoniecz** [comment]
> The general problem I see with tagging forests via survey is that it is usually a large area.

I thought that SC would be useful for mapping tiny areas that are easy to survey, hard to map tree type from aerial images and likely to not require splitting.

Is it possible in overpass to get areas below for example 1 square kilometer?

**Comment by PanderMusubi** [comment]
What is needed to move this issue forward?

**Comment by westnordost** [comment]
@matkoniecz Not sure, but it would be possible to calculate and filter that after download in Java code.

@PanderMusubi Someone willing to implement it

**Comment by matkoniecz** [comment]
> natural=scrub and barrier=hedge

Note the title of this issue. I am making second attempt at the PR that will support also `natural=wood` as obvious synonym of `landuse=forest`, but will not support natural=scrub and barrier=hedge. Primary reason is that separate translations are needed and that for tree covered area answering this quest is much more useful, as there is already some use of this tag.

For hedges it seems to be as useful as backrest quest for benches.

Feel free to create a separate issue for that, especially once this one will be closed.

I am planning to use https://commons.wikimedia.org/wiki/File:Juniper_needles.jpg https://commons.wikimedia.org/wiki/File:Leaf_of_common_tilia.jpg

In case of supporting leaft_type=palm: https://upload.wikimedia.org/wikipedia/commons/7/7e/2007-02-28_(57)_Palm_leaf,_Palmblatt.JPG

In case of supporting leafless: https://wiki.openstreetmap.org/wiki/File:Saguaro_jul04_38.jpg

With https://commons.wikimedia.org/wiki/File:Nadelnfichtenrp.jpg and https://commons.wikimedia.org/wiki/File:Populus_tremuloides_(5002308731).jpg as backup.

Other alternatives: https://commons.wikimedia.org/wiki/File:Quercus_robur.jpg https://en.wikipedia.org/wiki/File:Lisc_lipy.jpg  https://en.wikipedia.org/wiki/File:Abies_grandis_5359.JPG

> Did you see the trees/leaves in twemoji and emojione 2 icon sets?

Yes, but tree shape may not match leaf type and I see no icons in emojione for broad-leaved/needles distinction.

https://emojipedia.org/twitter/ has long list (better list is welcomed!), maybe I missed something but situation seems the same.


for a potential future search:

https://commons.wikimedia.org/wiki/Category:Leaves_on_white_background
https://commons.wikimedia.org/wiki/Category:Conifer_leaves
